### ISSUE TITLE
Some of multiple calls of `webR.evalR` do not resolve

### ISSUE THREAD

**Issue by kcf-jackson** [r-wasm/webr#255]
I am not sure if I am using it right, but when I try to make multiple `webR.evalR` calls sequentially / concurrently, it seems some of the `Promise` never resolve. Here is the code to reproduce the issue:

#### index.mjs
```{index.mjs}
import { WebR } from 'https://webr.r-wasm.org/latest/webr.mjs';
const webR = new WebR();
await webR.init();
globalThis.webR = webR;
```

Then in the browser console, 
```
let f = () => webR.evalR("a = 999;");
let g = () => webR.evalR("b = 999;");

f()
.then(x => console.log("f: ", x))
.catch(x => console.log(x));           // works fine

g()
.then(x => console.log("g: ", x))
.catch(x => console.log(x));           // works fine only if it is run standalone

Promise.all([f(), g()])        // doesn't resolve in browser console
.then(result => console.log("first:", result))
.catch(x => console.log(x));           

setTimeout(() => {
    Promise.all([f(), g()])        // a repeated call after a short time would work
    .then(result => console.log("second:", result))
    .catch(x => console.log(x));
}, 200)
```

Here is a screenshot showing the result:
![image](https://github.com/r-wasm/webr/assets/12783172/d30d8e4a-423f-41b9-9aed-ece3573d339c)

Thank you.

**Comment by georgestagg** [comment]
Thank you for reporting this!

Unfortunately, I can't seem to reproduce the same issue. I tried in a few browsers and the `Promises` all seem to resolve OK for me.

<img width="589" alt="Screenshot 2023-08-21 at 17 48 56" src="https://github.com/r-wasm/webr/assets/982548/0527dfe0-c38b-4427-89a5-e9b321964093">

Can you let me know the make (and version, if you know) of your web browser and operating system? Perhaps I can try on another machine or in a VM.

**Comment by kcf-jackson** [comment]
Thanks a lot @georgestagg for looking into this.

I am using 
- Chrome Version 115.0.5790.110, and
- Firefox 116.0.2 (Mozilla Firefox Snap for Ubuntu canonical-002 - 1.0)

on Ubuntu 22.04.2 LTS (Budgie 10.6.1). 

After seeing your reply, I have also tried 
- Chrome Version 111.0.5563.65 (Official Build), and 
- Firefox 116.0.3 

on Windows 11 Business 22H2 22621.2134 OS build.

After some more testing, it seems that on both OS the issue appeared when I served the page using VSCode Live Server (extension), and the issue went away when I switched back to serving with R using the template from the documentation. 

I will close this issue for now and reopen if the problem came back. Thanks again.

---

Thanks @bugzpodder for sharing. What happens when the webR processes step on each other? Does one of the results disappear? I tried using a queue to avoid the issue before, all the `Promise` did resolve correctly, but the program got into some locked states after awhile,  so I reverted my implementation.
